app/controllers/drivers_controller.rb

```diff
@@ -0,0 +1,75 @@
+class DriversController < ApplicationController
+  skip_before_action(:force_driver_sign_in, { :only => [:new_registration_form, :create] })
+  
+  def new_registration_form
+    render({ :template => "driver_sessions/sign_up.html.erb" })
+  end
+
+  def create
+    @driver = Driver.new
+          
+    @driver.email = params.fetch(:email, nil)
+                      
+    @driver.password = params.fetch(:password, nil)
+    @driver.password_confirmation = params.fetch(:password_confirmation, nil)
+                
+    @driver.username = params.fetch(:username, nil)
```

Comment:
What happens if there's a database default set for a column in this case? Does the value become `nil` or the db default?

Reply:
I don't think I'm able to determine if a particular column has a default value, since the migration hasn't run at this stage, but I can do something like

```ruby
@driver.username= params.fetch(:username, @driver.username)
```